src/openzaak/components/catalogi/admin/mixins.py

```diff
@@ -16,3 +18,21 @@ class ConceptAdminMixin(object):
     def get_fieldsets(self, request, obj=None):
         fieldsets = super().get_fieldsets(request, obj)
         return tuple(fieldsets) + ((_("Concept"), {"fields": ("concept",)}),)
+
+
+class PublishAdminMixin:
+    def response_post_save_change(self, request, obj):
```

Comment:
I see that you copied this code for Zaaktype with validation. That's not very DRY.

Why not add a new function like:

```python
def _publish_validation(...):
    pass  # default

def response_post_save_change(self, request, obj):
    if "_publish" in request.POST:
        # ...
        if not self._publish_validation():
            return
        # ...
```

And override `_publish_validation` with the code you added for Zaaktype?